File: app/src/main/java/com/krunal3kapadiya/popularmovies/data/adapter/TrailerRVAdapter.java
package com.krunal3kapadiya.popularmovies.data.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.krunal3kapadiya.popularmovies.R;
import com.krunal3kapadiya.popularmovies.data.model.Trailer;
import com.squareup.picasso.Picasso;

import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by Krunal on 8/5/2017.
 */

public class TrailerRVAdapter extends RecyclerView.Adapter<TrailerRVAdapter.ViewHolder> {

    private Context mContext;
    private ArrayList<Trailer> mTrailerList;
    private OnItemClick onItemClick;

    public TrailerRVAdapter(Context context, ArrayList<Trailer> trailerList) {
        mContext = context;
        mTrailerList = trailerList;
        onItemClick = (OnItemClick) context;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_trailer, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        //TODO add string here
        Picasso.with(mContext)
                .load(mContext.getString(R.string.youtube_image, mTrailerList.get(position).getKey()))
                .into(holder.mThumb);
    }

    @Override
    public int getItemCount() {
        return mTrailerList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.row_trailer_thumb)
        ImageView mThumb;

        @OnClick(R.id.row_trailer_thumb)
        void onClick() {
            onItemClick.onItemClick(getAdapterPosition());
        }

        ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }

    public interface OnItemClick {
        void onItemClick(int pos);
    }
}

File: app/src/main/java/com/krunal3kapadiya/popularmovies/data/adapter/TrailerRVAdapter.kt
package com.krunal3kapadiya.popularmovies.data.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.krunal3kapadiya.popularmovies.R
import com.krunal3kapadiya.popularmovies.data.model.Trailer
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.row_trailer.view.*
import java.util.*

class TrailerRVAdapter(private val mContext: Context, private val mTrailerList: ArrayList<Trailer>) : RecyclerView.Adapter<TrailerRVAdapter.ViewHolder>() {
    private val onItemClick: OnItemClick

    init {
        onItemClick = mContext as OnItemClick
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_trailer, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(mTrailerList)
    }

    override fun getItemCount(): Int {
        return mTrailerList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(mTrailerList: ArrayList<Trailer>) {
            with(mTrailerList) {
                Picasso.with(mContext)
                        .load(mContext.getString(R.string.youtube_image, mTrailerList[position].key))
                        .into(itemView.row_trailer_thumb)
            }

            itemView.row_trailer_thumb.setOnClickListener({ onItemClick.onItemClick(adapterPosition) })
        }
    }

    interface OnItemClick {
        fun onItemClick(pos: Int)
    }
}
